### ISSUE TITLE
Stage Plays in Better Fountain?

### ISSUE THREAD

**Issue by greggerca** [piersdeseilligny/betterfountain#117]
This extension is really great! 
I see it's possible to use Fountain's markup to do things like "ACT I" and "SCENE 1", but would it be easy to add common stage play syntax and formatting to the extension?

**Comment by piersdeseilligny** [comment]
@greggerca Thanks, i'm glad you like it! It may be possible, although I'm not at all familiar with stage play syntax/formatting. How different is it from screenplays? Do you have a link to a specific reference for it you would recommend?

**Comment by imaginariateatre** [comment]
Just the possibility to use a different font (i.e. Arial) would be great, at least here, people are weirded out by theater scripts in Courier. Also, by the indentation of the dialogue. Being able to adapt that, would be so great!

also: love the extension!

**Comment by piersdeseilligny** [comment]
@imaginariateatre thanks! and so you can actually already change the font :) just add the "Font:" title page key, as such:

```
Title: Something
Credit: By
Author: Someone
Font: Arial
```

For the indentation, it's also possible to change that, although in a hacky and undocumented way:

1. Open the following folder
        **Windows:** `%USERPROFILE%\.vscode\extensions\piersdeseilligny.betterfountain-1.8.2\out\pdf`
        **macOS:** `~/.vscode/extensions/piersdeseilligny.betterfountain-1.8.2/out/pdf`
        **Linux:** `~/.vscode/extensions/piersdeseilligny.betterfountain-1.8.2/out/pdf`
        
2. Open the `print.js` file

3. From there, you can edit the formatting used by the PDF renderer (these changes will not reflect in the live preview). So for example to change the character indentation, you can change the following lines, around line 40:
```js
character: {
    feed: 3.5,
    max: 33
},
```
to, for example:
```js
character: {
    feed: 1.5,
    max: A4_DEFAULT_MAX,
},
```


It's hacky and pretty ugly, but it should work. Obviously in the long run, this should be a more elegant feature (with for example custom presets) but I'm not sure if that will make it to the vscode extension, or if it will only be included in the standalone version.

**Comment by imaginariateatre** [comment]
thank you!!!! that worked perfect!

**Comment by greggerca** [comment]
> @greggerca Thanks, i'm glad you like it! It may be possible, although I'm not at all familiar with stage play syntax/formatting. How different is it from screenplays? Do you have a link to a specific reference for it you would recommend?

Hi there,
It took a while - but here's an example format we've used and created in another program. Stage plays have a few extra things, and the format isn't quite as rigorous. The title pages are very similar to a screenplay, but plays also include a dramatis personae page, author's notes, and stuff like that.

The [Dramatists Guild has a lot of the specifications laid out](https://www.dramatistsguild.com/script-formats). The one I'm attaching has the most similarities to the [Modern format](https://www.dramatistsguild.com/sites/default/files/2019-12/modernformat-New.pdf) and [Samuel French's format](https://www.dramatistsguild.com/sites/default/files/2020-01/General-SFI-Formatting-Guidelines-Complete.pdf). And the [BBC's format](http://downloads.bbc.co.uk/writersroom/scripts/stageus.pdf) is pretty similar as well.

The Fountain spec can sort of be tweaked to do the formatting correctly (adding a "." in front of ACT I or "~" in front of an action for a stage direction). I bet the PDF layout is harder though.

[Stageplay Template.pdf](https://github.com/piersdeseilligny/betterfountain/files/6054914/Stageplay.Template.pdf)


[Stageplay Template.fountain.txt](https://github.com/piersdeseilligny/betterfountain/files/6054911/Stageplay.Template.fountain.txt)

Thanks for looking into it! I know it's not the intent of Better Fountain. But the collaboration features and editor in VS Code seem like a nice experience for people!

**Comment by piersdeseilligny** [comment]
Thank you so much! That's really useful. I think this is definitely a feature that will make its way into the standalone version of betterfountain (maybe not on initial release though) - and maybe the vscode version too, but I'm not sure about that. I found a thing called [Downplay](https://github.com/barach/Downplay) but it's abandoned and doesn't seem to make a lot of sense, so I think the best would just be to have a preset for the PDF Renderer which has slightly different settings, and a slightly different flavour of fountain in order to add support for all the syntax involved in stage plays.

The question is now, what would have to change in the fountain syntax in order to make it compatible with stage plays?

Some quick thoughts/questions:

- Acts can continue using the "# Header" syntax (they just need to be printed differently)

- Scenes can start with "." (or maybe "##"?)

- Could stage directions not continue simply being an action block, just printed differently? "~" could then continue being used for lyrics.

**Comment by greggerca** [comment]
I think those suggestions would definitely work - I was surprised that the Fountain editor I exported that stage play out of didn't actually mark the "ACT I" with the # syntax, etc. I was expecting they'd use it as the Fountain spec suggested.

Scenes could certainly be designated by the "." syntax, and stage directions can definitely be an Action block - some playwrights use parentheses in stage directions, and some just do the indentation and italics. 

Parenthetical actions do exist in plays like (beat), (pause), or (nods) etc. - those are in between character "hints" to help the actor with intent and actual stage directions. Parenthetical things can appear within lines or in between lines.

I looked at Downplay as well, but that seems abandoned. It didn't seem complete to me.

Playwrights seem to be more flexible on the PDF format - they all look pretty similar, but as long as it's readable and all the pieces are clear, it seems to go well. Produced plays usually include more specific information pertaining to prop lists, stage layouts, and notations in the Title area about "first performed at XYZ theatre, with original cast of: <cast list>". That can be added in post somehow though. Those sections have a different numbering scheme to the rest of the play.

The most frequent new play request from directors is "can I have this in Word / Google Docs?" because they want to highlight (or suggest edits) during the first reading, rehearsal, and production. After production, the playwright will update the master script for future productions (if they liked the changes ;) ).

Also, distributing plays to actors by PDF or Word is fairly common. Actors tend to want to highlight their lines. Some do that on their own in the PDF, and some playwrights want to help by pre-highlighting the file(s) for each actor.

Screenwriters probably have a lot more revenue attached to their projects vs. plays :) so... you know, this is a lower priority.

A standalone Fountain app for Windows would be an excellent thing for screenwriters. The Mac has several, but Windows has very few.

Considering how a Fountain screenplay could be rendered in a Teleprompter app would also make for an interesting hand-off. There are plenty of YouTubers / Twitch content creators out there that would likely enjoy a nice writing experience that integrated with some teleprompter...

Thanks for looking into this!

**Comment by piersdeseilligny** [comment]
@greggerca Thanks for all of that! It's really helpful. What fountain editor were you using? Because most of the time, printing the sections is a setting that can be enabled (for stageplays, this would obviously be a default). That's how it works in betterfountain in any case:

![image](https://user-images.githubusercontent.com/42496508/110206451-9e79fc00-7e75-11eb-9fae-257822e27a15.png)


. And I hadn't at all thought about the teleprompter aspect. Obviously this isn't something which is within the scope of v.1.0 of bf-standalone, but it really shows how much potential there would be - either for more features or for extensions (which could for example add support for printing to different formats, like word or opendocument).

For the character highlighting, that's something which is already possible in betterfountain, with the "Export screenplay PDF with highlighted characters option" ;)

**Comment by imaginariateatre** [comment]
If I may add a few more thoughts... One of the things that changes the most between a script and a play (at least here) is how the dialogue is positioned with respect to the character's name. Though it's not mandatory, it is usually done on the same line, with a tab or colon to separate, and the name in general, is in bold, for example:

**MAN**:........What do you mean?
**WOMAN**:..That is what I mean

**MAN** ........What do you mean?
**WOMAN**....That is what I mean.

Thanks to your trick modifying print.js, I have been able to make the character name bold by changing the color of the text to dark grey, but no formatter allows character and dialog together. The most I get is:

**MAN**         
....................... What do you mean?
**WOMAN**   
.......................That is what I mean.

Which of course doubles the number of lines in the page, and the number of pages of the whole play.

I actually like better how fountain and scripts do it, it's clearer, but it would be very cool to be able to write in fountain but get it formatted like that. I don't know if I'm reaching for the moon here ...

Thanks for considering this a feature, btw, it's my dream come true ;-)

EDIT: Agh, had to put dots instead of spaces to demonstrate the tabs...

**Comment by piersdeseilligny** [comment]
Yeah that could definitely be a possibility (long-term). Although how does the alignment work for that? Because in your example, you've aligned it so that the dialogue is on the same indentation as the longest character name (`WOMAN`) :

```
MAN:    What do you mean
WOMAN:  That is what I mean
```

However what would happen if someone had a very long character name, for example? Would it then make _all_ the dialogue be really far off to the right??

**Comment by greggerca** [comment]
Hi there! 
The format imaginariatheatre is illustrating is something we've encountered in the published script / book format from Samuel French, Dramatists, etc. You definitely see that a lot in published works. But the script that Samuel French received from the playwright was more likely in the more traditional format before they printed it.

That compact book format saves a lot of space, and is super easy to carry around and highlight.
My suggestion would be to create it as a way to "render" (compile? :) ) or output that format option vs. as an a editing user experience.

From my experience, that compact format has two main drawbacks when producing / being in a new play:
1. Actors have little space for writing down blocking
2. New lines or word corrections supplied by the playwright during rehearsals are difficult to incorporate by hand in a very compact format
(of course, if there are enough changes, the playwright generates a new page in the script)

From the playwright perspective, the writing needs to be fast and fluid, it should be easy to rapidly write:
CHARACTER
Action / Stage Direction
Dialogue
(parenthesis) 

In software terms... parsing and structuring the playwright's input in the editor should be as predictable, trivial, and efficient as possible vs. attempting to format different elements on the same line. 

With a really predictable input structure, it should be easier to create the different outputs based on whatever layout is required. 

So if you want to output [Samuel French's play submission format](https://www.dramatistsguild.com/sites/default/files/2020-01/General-SFI-Formatting-Guidelines-Complete.pdf), you can. If you want a super compact format like imaginariatheatre would like, it should be pretty straightforward. 

But the two outputs should be possible from a single, good input / editor experience and formatting structure.

Of course, published play books are totally re-formatted once they go to publication. But I think it's important to make the editor itself easy to use, and capable to render the output format that makes sense for the PDF or Word doc's intended use.

We do a compromise on the formatting - we left justify the CHARACTER names so it's easier to read. It's kinda similar to the play book format, but a little roomier.

To answer some previous questions:
Character highlighting! Awesome - I missed that. This is a great example of what we're talking about - an output format that suits a specific purpose based on the same source doc.

I generated the example stage play Fountain doc from Highland. The source document started as a Final Draft template we created (based on a CeltX doc we had used). I also tried Fade In with similar results.

Thanks for the discussion!

**Comment by imaginariateatre** [comment]
> However what would happen if someone had a very long character name, for example? Would it then make _all_ the dialogue be really far off to the right??

You're right, it doesn't work to generate a general predictable input... I guess it's done like that, because you'll end up doing the format by hand and you know the names are short.

I've checked other published plays and everybody does it differently, some examples (this time I could write the tabs, so it shows as in the books):

Nick Hern Books:

HELEN. That's a holy flower. That keeps you away.
NICK. I like the smell of it.
  _(Offers it to her.)_ Here, you hold it then.
  HELEN _steps back._
HELEN. I want nothing from you.

Methuen Drama:

**Female Soldier** Alright.
**Haley** _yaps_.
**Male Soldier** Down the funny farn and chop inside yer
head till you could act normal.
**Female Soldier** You're beeding.
**Male Soldier** Course I'm fucking _ (_To_ **Haley**.) If you're
infected I'll have you.
_She gives him a cloth to mop the blood._

Heinemann Plays:

TOMMY   I never heard of the security forces checkin' anybody's
      age, name and address before they opened fire.
      _MAUREEN walks out. She passes ARTHUR who is clocking
      in._
ARTHUR  Mornin' sweetheart.
MAUREEN  Away an' jump.

I'm with greggerca that the input should be as easy and standard, that is what I love about writing in fountain.  But the thing is, most of the time I'm concerned with formatting the play is because I have to submit it to a contest, and each one establishes their rules, like i.e: "Times New Roman 12, double line spacing, max 60 pages".

Imagine being able to export my play in pdf just tweaking the configuration, without having to touch my original text to make a compact version, or generating multiple copies that mess with my head, just to format them differently, that would be great. One origin, multiple outputs. Even being able to save that configuration as "Contest A" or "Samuel French Submission Format", much like Highland 2 does it with its Document Templates, but much more configurable.

Thanks for the discussion, yeah! :-)

**Comment by piersdeseilligny** [comment]
I guess issues relating to the specific formatting of the plays can be resolved at a way later date, at which point all of this will be very useful :) The main good thing to know at this point, is that all of this can be achieved without really extending the fountain spec, it's all a rendering thing (and maybe some added convenience features, like automatically adding all characters to the "Cast of Characters" page at the beginning).

I think that the beta of bf-standalone, when it's finally released, won't initially include stage play formatting (the focus will be on making the damn best screenwriting software there is) but it will be something that can quite easily be added, maybe before the 1.0 Release.

> Imagine being able to export my play in pdf just tweaking the configuration, without having to touch my original text to make a compact version, or generating multiple copies that mess with my head, just to format them differently, that would be great. One origin, multiple outputs.

Definitely the aim :) rendering presets, loosely inspired from the way DaVinci Resolve does it for video, will be one of the core features of bf-standalone. That's what's so amazing about fountain; it's 100% human-readable, with a very gentle learning curve, but can be rendered however you want. It's really ideal for screenplays, stageplays, maybe even AV Scripts and novels, etc... 


Actually, it would even be very easy to switch the same document between a stageplay and a screenplay :o
